api/python/tests/integration/test_packages.py

```diff
@@ -18,6 +19,49 @@ def mock_make_api_call(operation_name):
         return parsed_response
     raise NotImplementedError
 
+def test_build(tmpdir):
+    """Verify that build dumps the manifest to appdirs directory."""
+    # Patch appdirs to use a temp dir.
+    appdirs.user_data_dir = lambda x: os.path.join(tmpdir, x)
+
+    new_pkg = Package()
+
+    # Create a dummy file to add to the package.
+    test_file = os.path.join(tmpdir, 'bar')
+    with open(test_file, "w") as fd:    
+        fd.write(test_file)
+
+    # Build a new package into the local registry.
+    new_pkg = new_pkg.set('foo', test_file)
+    top_hash = new_pkg.build("Test")
+
+    # Verify manifest is registered by hash.
+    out_path = os.path.join(appdirs.user_data_dir("quilt"), "packages", top_hash)
+    with open(out_path) as fd:
+        pkg = Package.load(fd)
+        assert "file://" + test_file \
```

Comment:
Well, it's a URI, so it's not that simple... You technically need to quote it. If you dare to try it on Windows, it's even crazier:
```
E           AssertionError: assert 'file://C:\\u...t_build0\\bar' == 'file:///C:/us...st_build0/bar'
E             - file://C:\users\dima\Temp\pytest-of-dima\pytest-1\test_build0\bar
E             ?          ^     ^    ^    ^              ^        ^           ^
E             + file:///C:/users/dima/Temp/pytest-of-dima/pytest-1/test_build0/bar
E             ?        +  ^     ^    ^    ^              ^        ^           ^
```
But if you use `Path`, you should be able to just call `test_file.resolve().as_uri()`.

Alternatively, just call `pkg.get(...)` and not worry about the exact URL value.